### DISCUSSION TITLE
Non-Error exception captured with keys: [...]

### DISCUSSION THREAD

**Discussion by asindhu** [getsentry/sentry-react-native#3320] [Q&A]
Using Sentry on react-native with Expo, totally new to Sentry and relatively new to all other pieces of this stack. In some places in my code I am trying to use Sentry to capture errors that are returned from various frameworks, in other words they are not native JS Error objects. This is how it's invoked in my code:

`var error = { some error returned from a 3rd party library that's not a native JS Error }`
`Sentry.Native.captureException(error);`

From the Sentry documentation itself, it says pretty clearly that capturing non-Error objects should be fine, and that the only issue is that you may be missing a stack trace for those:

> You can pass an Error object to captureException() to get it captured as event. It's also possible to pass non-Error objects and strings, but be aware that the resulting events in Sentry may be missing a stack trace.

But what I see when one of these events triggers, in the Sentry dashboard, is only the following:

> Non-Error exception captured with keys: code, details, hint, message

It gives me this message, which is expected -- yes, my code captured a non-Error object with exactly those properties. But the actual object doesn't seem to be logged anywhere. I have the keys, but not the values!

Is there a different/better way to fully capture non-Error objects in Sentry? I don't care about the fact that the stack trace is missing, but I do care about actually seeing the whole object (keys and properties) in the Sentry dashboard for the objects I'm capturing.

Sorry in advance if this is a very basic question. I spent a lot of time going through documentation and searching online for an answer before posting this. Thanks!

**Comment by krystofwoldrich** [comment]
Hi @asindhu,
I'm sorry for the delayed answer. Great question.

You can add the captured object values in the `beforeSend` callback.

```js
Sentry.init({
  beforeSend: (event: Sentry.Event, hint: EventHint) => {
	// check if the originalException is your object and add the data as extras to the Sentry Event
    if (typeof hint.originalException === 'object' && hint.originalException != null &&  'key' in hint.originalException) {
      event.extra = {
        ...event.extra,
        ...hint.originalException,
      }
    }
    return event;
  },
});
```

**Reply by asindhu** [reply]
Great, thanks very much! Will give this a try. Do you happen to have a pointer to any further documentation on this? Did a quick search but couldn't find much other than [this page](https://docs.sentry.io/platforms/node/enriching-events/event-processors/).
